File: TestDoubles/Java/SpySampleTest.java
import org.junit.*;
import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.*;

public class SpySampleTest
{
    @Test
    public void test_with_spy() 
    {
        HikerSpy spy = new HikerSpy();
        Hiker douglas = new Hiker(spy);

        assertThat(douglas.answerFor(10), is(90));
        assertThat(spy.multiplierWasCalled, is(true));
    }

    class HikerSpy extends HikerHelper
    {
        public boolean multiplierWasCalled = false;

        @Override
        public int multiplier() 
        {
            multiplierWasCalled = true;
            return super.multiplier();
        }
    }
}

File: TestDoubles/C#/SpySampleTest.cs
using System;
using NUnit.Framework;

[TestFixture]
public class SpySampleTest
{
    [Test]
    public void test_with_spy() 
    {
        HikerSpy spy = new HikerSpy();
        Hiker douglas = new Hiker(spy);

        Assert.That(douglas.AnswerFor(6), Is.EqualTo(42));
        Assert.That(spy.multiplierWasCalled, Is.EqualTo(true));
    }


    class HikerSpy: RealHelper
    {
        public bool multiplierWasCalled = false;

        public override int GetMultiplier() 
        {
            multiplierWasCalled = true;
            return base.GetMultiplier();
        }
    }
}
